Modules/Components/Examples/include/selxIdentityTransformRegistrationComponent.hxx

```diff
@@ -0,0 +1,121 @@
+IdentityTransformRegistrationComponent< Dimensionality, TPixel >::~IdentityTransformRegistrationComponent()
+{
+}
+
+
+template< int Dimensionality, class TPixel >
+int
+IdentityTransformRegistrationComponent< Dimensionality, TPixel >::Accept( typename itkImageFixedInterface< Dimensionality, TPixel >::Pointer component )
+{
+  // Since SuperElastix is an itk filter in a pipeline, we need to set the domain of the resulting deformation field before running the registration.
+  // As done in pairwise registration, typically, we use the domain of the fixed image for that.
+  auto fixed = component->GetItkImageFixed();
+  fixed->UpdateOutputInformation();
+  auto size = fixed->GetLargestPossibleRegion().GetSize();
+  this->m_DisplacementField->SetRegions(size);
+
+  auto spacing = fixed->GetSpacing();
+  this->m_DisplacementField->SetSpacing(spacing);
+
+  auto origin = fixed->GetOrigin();
+  this->m_DisplacementField->SetOrigin(origin);
+
+  auto direction = fixed->GetDirection();
+  this->m_DisplacementField->SetDirection(direction);
+
+  this->m_ImageFixedInterface = component;
+  return 0;
+}
+
+
+template< int Dimensionality, class TPixel >
+int
+IdentityTransformRegistrationComponent< Dimensionality, TPixel >::Accept( typename itkImageMovingInterface< Dimensionality, TPixel >::Pointer component )
+{
+  this->m_ImageMovingInterface = component;
+  return 0;
+}
+
+
+template< int Dimensionality, class TPixel >
+typename IdentityTransformRegistrationComponent< Dimensionality, TPixel >::DisplacementFieldImageType::Pointer
+IdentityTransformRegistrationComponent< Dimensionality, TPixel >::GetDisplacementFieldItkImage()
+{
+
+  return this->m_DisplacementField;
+}
+
+template< int Dimensionality, class TPixel >
+void
+IdentityTransformRegistrationComponent< Dimensionality, TPixel >::RunRegistration( void )
+{
+  auto fixed = this->m_ImageFixedInterface->GetItkImageFixed();
+  auto moving = this->m_ImageMovingInterface->GetItkImageMoving();
+  
+  fixed->SetRequestedRegionToLargestPossibleRegion();
+  fixed->Update();
+  // Raw pixel data is accessible by GetBufferPointer()
+  moving->SetRequestedRegionToLargestPossibleRegion();
+  moving->Update();
+  // Raw pixel data is accessible by GetBufferPointer()
+  
+  // Here you can implement your registration algorithm using the fixed and moving image.
+  // For an Identity Transform it is just creating a vector field with zeros.
+  this->m_DisplacementField->Allocate();
+}
+
+
+template< int Dimensionality, class TPixel >
+bool
+IdentityTransformRegistrationComponent< Dimensionality, TPixel >
+::MeetsCriterion( const CriterionType & criterion )
+{
+  bool hasUndefinedCriteria( false );
+  bool meetsCriteria( false );
+  auto status = CheckTemplateProperties( this->TemplateProperties(), criterion );
+  if( status == CriterionStatus::Satisfied )
+  {
+    return true;
+  }
+  else if( status == CriterionStatus::Failed )
+  {
+    return false;
+  } // else: CriterionStatus::Unknown
+
```

Comment:
"Get properties here"

Reply:
Yes, I will extend this as an example how to use the component its settings.